src/sentry/api/utils.py

```diff
@@ -346,3 +366,68 @@ def get_auth_api_token_type(auth: object) -> str | None:
     if is_api_key_auth(auth):
         return "api_key"
     return None
+
+
+# NOTE: This duplicates OrganizationEventsEndpointBase.handle_query_errors
+# TODO: move other references over to this implementation rather than the organization_events implementation
+@contextmanager
+def handle_query_errors(self) -> Generator[None, None, None]:
```

Comment:
Could we move this to a format that's a little easier to extend and add on? Something like (pseudo code):
```
from contextlib import contextmanager
from typing import Type, TypeVar

T = TypeVar('T', bound='ErrorHandler')

class NoArgHandler:
    @staticmethod
    def handle_error(error: Exception) -> None:
        # Add your logic here
        print(f"Handling NoArg error: {error}")
        return

class ErrorHandler:
    error_handlers = {
        "NoArg": NoArgHandler,
        # Add more handlers as needed
    }

    @classmethod
    @contextmanager
    def handle(cls: Type[T], error: Exception) -> None:
        handler_type = cls.error_handlers.get(type(error).__name__, None)
        if handler_type is None:
            raise Exception("No handler for this unexpected error")

        handler_instance = handler_type()
        try:
            yield
        except Exception as e:
            handler_instance.handle_error(e)

# Example usage:
with ErrorHandler.handle(ValueError("Test ValueError")):
    # Your code that might raise an exception
    x = int("abc")  # This will raise a ValueError

```